lib/Runtime/PlatformAgnostic/Platform/Common/HiResTimer.cpp

```diff
@@ -64,28 +64,33 @@ namespace DateTime
                                             utc_tm.tm_mday - 1, milliseconds);
     }
 
+// Important! When you update 5ms below to any other number, also update test/Date/xplatInterval.js 0->5 range
 #define INTERVAL_FOR_TICK_BACKUP 5
     double HiResTimer::GetSystemTime()
     {
         ULONGLONG current = GetTickCount64();
         ULONGLONG diff = current - data.cacheTick;
 
-        if (diff >= INTERVAL_FOR_TICK_BACKUP) // max *ms to respond system time changes
+        if (diff <= data.previousDifference || diff >= INTERVAL_FOR_TICK_BACKUP) // max *ms to respond system time changes
         {
             double currentTime = GetSystemTimeREAL();
 
             // in case the system time wasn't updated backwards, and cache is still beyond...
             if (currentTime >= data.cacheSysTime && currentTime < data.cacheSysTime + INTERVAL_FOR_TICK_BACKUP)
             {
+                data.previousDifference = (ULONGLONG) -1; // Make sure next request won't use cache
```

Comment:
>// Make sure next request won't use cache [](start = 58, length = 41)

Just to understand this, what's the case that this time it didn't use cache but next time it uses cache? Is it because of tick cycle?